apps/process_tpstream.cxx

```diff
@@ -345,92 +313,37 @@ int main(int argc, char const *argv[])
       //
       // TC Processing
       //
-      // Create the output buffer
-      std::vector<triggeralgs::TriggerCandidate> tc_buffer;
-
-      // Loop over TPs
-      size_t n_tcs = 0;
-
-      for( const auto& ta : ta_buffer ) {
-
-        // if ( tp_filter(tp) ){
-        //   if(!quiet)
-        //       fmt::print("  TP filtered out!");
-        //   continue;
-        // }
-
-        (*tc_maker)(ta, tc_buffer);
-
-        if (n_tcs != tc_buffer.size()) {
-          if (!quiet){
-            for( size_t i=n_tcs; i<tc_buffer.size(); ++i){
-              const auto& tc = tc_buffer[i];
-                fmt::print("  + {} tc_s={}, tc_e={} | ta_s={} ta_s-tc_s={}\n", i, tc.time_start, tc.time_end, ta.time_start, (ta.time_start-tc.time_start) );
-            }
-          }
-          n_tcs = tc_buffer.size();
-        }
-      }
-
-      // Count how many TCs were generated
-      if (!quiet)
-        fmt::print("  tc_buffer.size() = {}\n", tc_buffer.size());
 
-      // Their size
-      size_t tc_payload_size(0);
-      for ( const auto& tc : tc_buffer ) {
-        tc_payload_size += triggeralgs::get_overlay_nbytes(tc);
-      }
+      std::vector<triggeralgs::TriggerActivity> ta_buffer = ta_emulator->get_last_output_buffer();
+      const auto tc_start = std::chrono::steady_clock::now();
+      std::unique_ptr<daqdataformats::Fragment> tc_frag = tc_emulator->emulate(ta_buffer);
+      const auto tc_end = std::chrono::steady_clock::now();
```

Comment:
Likewise here, would it be possible to have a for loop that feeds the tc emulator one-by-one, and include all the `continue`, amd `tc_diff` in that for loop? That will give us the time between the last TA pushed into TCMaker and the TC made, which is pretty much what we want.

In the further future we can even have different definitions of a "latency", but  that would work.